IdentityCore/src/oauth2/MSIDOauth2Factory.m

```diff
@@ -411,6 +416,12 @@ - (BOOL)fillAppMetadata:(MSIDAppMetadataCacheItem *)metadata
     return YES;
 }
 
+- (BOOL)doesResponseHaveBoundAppRefreshToken:(MSIDTokenResponse *)response
+{
+    return ![NSString msidIsStringNilOrBlank:response.boundAppRefreshTokenDeviceId] && ([response.additionalServerInfo[MSID_REFRESH_TOKEN_TYPE] isEqualToString:MSID_REFRESH_TOKEN_TYPE_BOUND_APP_RT] ||
```

Comment:
Should we switch the order?
`[response.additionalServerInfo[MSID_REFRESH_TOKEN_TYPE] isEqualToString:MSID_REFRESH_TOKEN_TYPE_BOUND_APP_RT]` in case there is no value for the key`MSID_REFRESH_TOKEN_TYPE` and calling `isEqualToString ` from null will crash

Reply:
Right, the dictionary returns nil instead of null when there is no value. Resolving